File: app/src/test/java/com/fractalwrench/crazycats/resource/ResourceFileReaderTest.java
package com.fractalwrench.crazycats.resource;

import org.junit.Before;
import org.junit.Test;

import java.io.IOException;

import static junit.framework.Assert.assertEquals;

public class ResourceFileReaderTest {

    private static final String EXPECTED_OUTPUT = "test";
    private static final String FILENAME = "test.txt";
    private StringResourceFileReader reader;

    @Before
    public void setUp() throws Exception {
        reader = new StringResourceFileReader();
    }

    @Test
    public void readResourceAsString() throws Exception {
        assertEquals(EXPECTED_OUTPUT, reader.readResourceAsString(FILENAME));
    }

    @Test(expected = IllegalArgumentException.class)
    public void nullName() throws Exception {
        reader.readResourceAsString(null);
    }

    @Test(expected = IOException.class)
    public void resourceNotFound() throws Exception {
        reader.readResourceAsString("totally-legit.jpg");
    }

}

File: app/src/test/java/com/fractalwrench/crazycats/resource/ResourceFileReaderTest.kt
package com.fractalwrench.crazycats.resource

import junit.framework.Assert.assertEquals
import org.junit.Before
import org.junit.Test
import java.io.IOException

class ResourceFileReaderTest {
    private var reader: StringResourceFileReader? = null

    @Before
    @Throws(Exception::class)
    fun setUp() {
        reader = StringResourceFileReader()
    }

    @Test
    @Throws(Exception::class)
    fun readResourceAsString() {
        assertEquals(EXPECTED_OUTPUT, reader!!.readResourceAsString(FILENAME))
    }

    @Test(expected = IOException::class)
    @Throws(Exception::class)
    fun resourceNotFound() {
        reader!!.readResourceAsString("totally-legit.jpg")
    }

    companion object {

        private val EXPECTED_OUTPUT = "test"
        private val FILENAME = "test.txt"
    }

}